fastly/block_fastly_service_v1_settings.go

```diff
@@ -19,15 +19,12 @@ func (h *SettingsServiceAttributeHandler) Process(d *schema.ResourceData, latest
 	opts := gofastly.UpdateSettingsInput{
 		ServiceID:      d.Id(),
 		ServiceVersion: latestVersion,
+		DefaultHost:    gofastly.String(d.Get("default_host").(string)),
```

Comment:
@smaeda-ks @bengesoff 

I might just be confused about the problem but is the change required here actually to update go-fastly so that the `DefaultHost` is no longer a pointer reference (and to not `omitempty`), then we update the terraform provider to use the new go-fastly release?

https://github.com/fastly/go-fastly/blob/6bf96a5a4c2ba87230e30ddd8683626aa0828bc4/fastly/settings.go#L57

That way the terraform logic can stay the same and it means that if it doesn't execute the setting of `opts.DefaultHost`, then go-fastly would use a zero value of `""` for that field.

Otherwise we're trying to side-step the go-fastly implementation when that implementation appears to be incorrect.

Reply:
I think I see what you're saying, but might it be relevant to update other Settings and leave `DefaultHost` set how it is? And for that you'd need a pointer and `omitempty` so you can have a `nil` pointer and omit it from the update?

Reply:
I think go-fastly is fine. `general.default_host` is an optional parameter so when there's no change the client doesn't need to send this parameter at all, and `omitempty` should be there.

But in order to "delete" this setting, the client must send `...&general.default_host=&...` as a query parameter. Right now, when `default_host` is set to `""` or deleted from TF file, we're just not sending this parameter at all, and has no effect on that value.

Reply:
Ah OK I see 👍🏻